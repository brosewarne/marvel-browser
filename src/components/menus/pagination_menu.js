import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import './menu.css';

/**
 * Render a PaginationMenu for navigating through each set of characters.
 * @returns {Node|null} - The pagination menu, or null if no characters have been loaded.
 */
export const PaginationMenu = ({ totalPages, startsWith }) => {
    if (!totalPages) {
        return null;
    }

    const pageTiles = [];
    for (let i = 0; i < totalPages; i++) {
        const href = `/characters/${startsWith}/${i + 1}`;
        pageTiles.push(
            <div className="menuTile" key={`pagination_div_${i}`}>
                <Link href={href} to={href} key={`pagination_${i}`}>{i + 1}</Link>
            </div>
        );
    }
    return (
        <div className="menu">
            {pageTiles}
        </div>
    );
};

PaginationMenu.propTypes = {
    totalPages: PropTypes.number.isRequired,
    startsWith: PropTypes.string.isRequired
};
