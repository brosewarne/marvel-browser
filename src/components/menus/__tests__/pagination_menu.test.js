import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { PaginationMenu } from '../pagination_menu';

configure({ adapter: new Adapter() });

describe('<PaginationMenu />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <PaginationMenu totalPages={15} startsWith="A" />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('doesn\'t render if totalPages is 0', () => {
        const wrapper = shallow(
            <PaginationMenu totalPages={0} startsWith="A" />
        );
        expect(toJson(wrapper)).toBeFalsy();
    });
});
