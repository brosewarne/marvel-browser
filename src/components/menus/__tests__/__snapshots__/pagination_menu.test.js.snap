// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationMenu /> renders correctly 1`] = `
<div
  className="menu"
>
  <div
    className="menuTile"
    key="pagination_div_0"
  >
    <Link
      href="/characters/A/1"
      key="pagination_0"
      replace={false}
      to="/characters/A/1"
    >
      1
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_1"
  >
    <Link
      href="/characters/A/2"
      key="pagination_1"
      replace={false}
      to="/characters/A/2"
    >
      2
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_2"
  >
    <Link
      href="/characters/A/3"
      key="pagination_2"
      replace={false}
      to="/characters/A/3"
    >
      3
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_3"
  >
    <Link
      href="/characters/A/4"
      key="pagination_3"
      replace={false}
      to="/characters/A/4"
    >
      4
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_4"
  >
    <Link
      href="/characters/A/5"
      key="pagination_4"
      replace={false}
      to="/characters/A/5"
    >
      5
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_5"
  >
    <Link
      href="/characters/A/6"
      key="pagination_5"
      replace={false}
      to="/characters/A/6"
    >
      6
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_6"
  >
    <Link
      href="/characters/A/7"
      key="pagination_6"
      replace={false}
      to="/characters/A/7"
    >
      7
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_7"
  >
    <Link
      href="/characters/A/8"
      key="pagination_7"
      replace={false}
      to="/characters/A/8"
    >
      8
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_8"
  >
    <Link
      href="/characters/A/9"
      key="pagination_8"
      replace={false}
      to="/characters/A/9"
    >
      9
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_9"
  >
    <Link
      href="/characters/A/10"
      key="pagination_9"
      replace={false}
      to="/characters/A/10"
    >
      10
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_10"
  >
    <Link
      href="/characters/A/11"
      key="pagination_10"
      replace={false}
      to="/characters/A/11"
    >
      11
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_11"
  >
    <Link
      href="/characters/A/12"
      key="pagination_11"
      replace={false}
      to="/characters/A/12"
    >
      12
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_12"
  >
    <Link
      href="/characters/A/13"
      key="pagination_12"
      replace={false}
      to="/characters/A/13"
    >
      13
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_13"
  >
    <Link
      href="/characters/A/14"
      key="pagination_13"
      replace={false}
      to="/characters/A/14"
    >
      14
    </Link>
  </div>
  <div
    className="menuTile"
    key="pagination_div_14"
  >
    <Link
      href="/characters/A/15"
      key="pagination_14"
      replace={false}
      to="/characters/A/15"
    >
      15
    </Link>
  </div>
</div>
`;
